src/python/pants/core/util_rules/archive.py

```diff
@@ -64,14 +64,14 @@ async def create_archive(request: CreateArchive) -> Digest:
     )
     file_list_file_digest = await Get(Digest, CreateDigest([file_list_file]))
     files_digests = [file_list_file_digest, request.snapshot.digest]
+    input_digests = []
```

Comment:
When using `append` on this I had to give it a type annotation inside the branch. That made the declarations in the two branches conflict. If I move it out here the type inference in MyPy is OK with append again since it's shared in both branches.

Reply:
Fine with me! Sometimes we have to do a little dance around mypy's sensitivities...